Add express route for student login and mongoose schema files

sasha-klnd opened the pull request:
#13

JenLys committed:
html to react

JenLys committed:
HTML code added as a comment

JenLys committed:
added

JenLys committed:
added navbar

JenLys committed:
fixed video

JenLys committed:
css

sasha-klnd committed:
Rebase Task.04 with main

sasha-klnd committed:
Add express route for student login and mongoose schema files

Vinuyans commented on server/routes/api.js:
```diff
@@ -1,10 +1,27 @@
-import express from "express";
+import express, { urlencoded } from "express";
 // import db from "../db/connection.js"
 import { ObjectId } from "mongodb";
```
You wouldn't need this line here

Vinuyans commented on server/db/schemas/course-schema.js:
```diff
@@ -0,0 +1,14 @@
+        course_id: String,
+        number: String,
+        dept: String,
+        prof: User.schema,
```
this should just be a ID to a user to save space and avoid having the whole User embedded in

Vinuyans reviewed (commented):
Great job learning Schemas 😄

JenLys committed:
changing branch name, updating navbar, fixing Welcome routes

JenLys committed:
update imports

JenLys committed:
mp4 converted to webm

JenLys committed:
slight ui css changes

Marian-Hristov committed:
Added mongoose instead of mongodb as db client

Marian-Hristov committed:
remove .env file

Marian-Hristov committed:
Add nodemon and fix db connection

Vinuyans commented:
You would want to change the names of the schemas from user_schemas to just user since they're already in the schemas folder

sasha-klnd committed:
Rebase Task.04 with main

sasha-klnd committed:
Update schema files and connection.js

Updates data types and mongoose connection for all the schema files. Updated connection.js to only work with mongoose, rather than mongo

sasha-klnd committed:
Merge branch 'Task.04' of https://github.com/Vinuyans/Mega_Chug_Lords-SOEN341_Project_F24 into Task.04
